ts/session/apis/open_group_api/sogsv3/sogsApiV3.ts

```diff
@@ -0,0 +1,501 @@
+        }
+        await Data.removeMessage(id);
+      })
+    );
+    //
+  } catch (e) {
+    window?.log?.warn('handleDeletions failed:', e);
+  }
+  return exceptDeletion;
+};
+
+const handleMessagesResponseV4 = async (
+  messages: Array<OpenGroupMessageV4>,
+  serverUrl: string,
+  subrequestOption: SubRequestMessagesType,
+  roomIdsStillPolled: Set<string>
+) => {
+  if (!subrequestOption || !subrequestOption.messages) {
+    window?.log?.error('handleBatchPollResults - missing fields required for message subresponse');
+    return;
+  }
+
+  try {
+    const { roomId } = subrequestOption.messages;
+
+    if (!roomIdsStillPolled.has(roomId)) {
+      window.log.info(`handleMessagesResponseV4: we are no longer polling for ${roomId}: skipping`);
+      return;
+    }
+    const convoId = getOpenGroupV2ConversationId(serverUrl, roomId);
+
+    const roomInfos = await getRoomAndUpdateLastFetchTimestamp(convoId, messages);
+    if (!roomInfos || !roomInfos.conversationId) {
+      return;
+    }
+
+    if (!isArray(messages)) {
+      window.log.warn("handleMessagesResponseV4: didn't get an object from batch poll");
+      return;
+    }
+
+    // Incoming messages for sogs v3 have a timestamp in seconds and not ms.
+    // Session works with timestamp in ms, for a lot of things, so first, lets fix this.
+
+    const messagesWithMsTimestamp = messages.map(m => ({
+      ...m,
+      posted: Math.floor(m.posted * 1000),
+    }));
+
+    const messagesWithoutDeleted = await handleSogsV3DeletedMessages(
+      messagesWithMsTimestamp,
+      serverUrl,
+      subrequestOption.messages.roomId
+    );
+
+    const messagesWithValidSignature = await filterOutMessagesInvalidSignature(
+      messagesWithoutDeleted
+    );
+    // we do a first check with blinded ids. Looking to filter out messages we already received from that blinded id.
+    const messagesFilteredBlindedIds = await filterDuplicatesFromDbAndIncomingV4(
+      messagesWithValidSignature
+    );
+
+    const roomDetails: OpenGroupRequestCommonType = pick(roomInfos, 'serverUrl', 'roomId');
+    // then we try to find matching real session ids with the blinded ids we have.
+    // this is where we override the blindedId with the real one in case we already know that user real sessionId
+
+    // tslint:disable: prefer-for-of
+    const messagesWithResolvedBlindedIdsIfFound = [];
+    for (let index = 0; index < messagesFilteredBlindedIds.length; index++) {
+      const newMessage = messagesFilteredBlindedIds[index];
+      const unblindedIdFound = getCachedNakedKeyFromBlindedNoServerPubkey(newMessage.session_id);
+
+      // override the sender in the message itself if we are the sender
+      if (unblindedIdFound && UserUtils.isUsFromCache(unblindedIdFound)) {
+        newMessage.session_id = unblindedIdFound;
+      }
+      messagesWithResolvedBlindedIdsIfFound.push(newMessage);
+    }
+    const dedupedUnblindedMessages = await filterDuplicatesFromDbAndIncomingV4(
```

Comment:
Why would this need to be run again since it was already filtered?

Reply:
so, I think this was because at first I was overriding the sender of a blinded message with its real id if I had it, locally


Now that we do not do this anymore, this line can be safely removed.